spec/jobs/cron_job/refresh_blog_posts_from_headway_job_spec.rb

```diff
@@ -0,0 +1,32 @@
+RSpec.describe CronJob::RefreshBlogPostsFromHeadwayJob do
+  context "when Headway is available" do
+    before do
+      stub_request(:get, BlogPost::HEADWAY_URL).to_return(body: file_fixture("headway_home.html"))
+    end
+
+    it "parses posts and passes them" do
+      expect(BlogPost).to receive(:refresh_from_posts).with(satisfy do |post_hashes|
+        expect(post_hashes.pluck(:title)).to contain_exactly("Mots de passe forts obligatoires", "Simplification du formulaire des motifs", "Documentation")
+      end)
+      described_class.new.perform
+    end
+  end
+
+  context "when Headway is down" do
+    before do
+      stub_request(:get, BlogPost::HEADWAY_URL).to_timeout
+    end
+
+    it "warns Sentry and retries the job" do
+      expect(enqueued_jobs).to be_empty
+      expect(sentry_events).to be_empty
+
+      described_class.perform_now
```

Comment:
nitpick : éventuellement rajouter cet expect qui me semble important

```suggestion
      expect(BlogPost).not_to receive(:refresh_from_posts) 
      described_class.perform_now
```

Reply:
J'ai fait comme ça : 63529c24